packages/react-components/react-migration-v8-v9/src/components/Checkbox/shimCheckboxProps.ts

```diff
@@ -0,0 +1,86 @@
+import { ICheckboxProps } from '@fluentui/react';
+import { CheckboxOnChangeData, CheckboxProps } from '@fluentui/react-components';
+
+import { getHTMLAttributes } from '../utils';
+
+// https://react.fluentui.dev/?path=/docs/concepts-migration-from-v8-components-checkbox-migration--page
+// https://github.com/microsoft/fluentui/blob/master/packages/react/src/components/Checkbox/Checkbox.types.ts
+// https://github.com/microsoft/fluentui/blob/103b8977f8d5f8dd8c430bab46ff5308a2c76371/packages/react-components/react-checkbox/src/components/Checkbox/Checkbox.types.ts
+
+export const shimCheckboxProps = (props: ICheckboxProps): CheckboxProps => {
+  const {
+    ariaDescribedBy,
+    ariaLabel,
+    ariaLabelledBy,
+    ariaPositionInSet,
+    ariaSetSize,
+    boxSide,
+    checked,
+    componentRef,
+    defaultChecked,
+    defaultIndeterminate,
+    disabled,
+    indeterminate,
+    inputProps,
+    name,
+    onChange: onChangeV8,
+    required,
+    title,
+  } = props;
+
+  const onChange =
+    onChangeV8 &&
+    ((ev: React.ChangeEvent<HTMLInputElement>, data: CheckboxOnChangeData) => {
+      return onChangeV8(ev, data.checked as boolean);
+    });
+
+  const V9Props: Partial<CheckboxProps> = {
+    ref: componentRef,
+    checked: checked || indeterminate,
+    defaultChecked: defaultChecked || defaultIndeterminate,
+
+    labelPosition: boxSide === 'end' ? 'before' : 'after',
+    onChange,
+  } as Partial<CheckboxProps>;
+
+  return {
+    ...inputProps, // This inputProps is specific for the input element, and the html attributes are also used here instead of props.
+    'aria-describedby': ariaDescribedBy,
```

Comment:
Question: Is it duplicate to write aria-describedby here when there is the getHTMLAttributes that includes it too?

Reply:
It should not be duplicated, as the "aria-describedby" is defined as ariaDescribedBy in V8, need to be translated to the origin html.